lib/ansible/parsing/vault/util.py

```diff
@@ -0,0 +1,94 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+from binascii import Error as BinasciiError
+from binascii import unhexlify
+
+from ansible import constants as C
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_text
+
+b_HEADER = b'$ANSIBLE_VAULT'
+
+
+class AnsibleVaultError(AnsibleError):
+    pass
+
+
+class AnsibleVaultPasswordError(AnsibleVaultError):
+    pass
+
+
+class AnsibleVaultFormatError(AnsibleError):
+    pass
+
+
+def parse_vaulttext_envelope(b_vaulttext_envelope, default_vault_id=None, filename=None):
+    """
+    Parse the vaulttext envelope
+
+    When data is saved, it has a header prepended and is formatted into 80
+    character lines.  This method extracts the information from the header
+    and then removes the header and the inserted newlines.  The string returned
+    is suitable for processing by the Cipher classes.
+
+    :arg b_vaulttext_envelope: byte str containing the data from a save file
+    :arg default_vault_id: The vault_id name to use if the vaulttext does not provide one.
+    :arg filename: The filename that the data came from.  This is only
+        used to make better error messages in case the data cannot be
+        decrypted. This is optional.
+    :returns: A tuple of byte str of the vaulttext suitable to pass to parse_vaultext,
+        a byte str of the vault format version,
+        the name of the cipher used, and the vault_id.
+    :raises: AnsibleVaultFormatError: if the vaulttext_envelope format is invalid
+    """
+    # used by decrypt
+    default_vault_id = default_vault_id or C.DEFAULT_VAULT_IDENTITY
+
+    try:
+        return _parse_vault_text_envelope(b_vaulttext_envelope, default_vault_id)
+    except Exception as exc:
+        msg = "Vault envelope format error"
+        if filename:
+            msg += f' in {filename}'
+        msg += f': {exc}'
+        raise AnsibleVaultFormatError(msg)
+
+
+def _parse_vault_text_envelope(b_vault_text_envelope, default_vault_id=None):
+    b_tmp_data = b_vault_text_envelope.splitlines()
+    b_tmp_header = b_tmp_data[0].strip().split(b';')
+
+    b_version = b_tmp_header[1].strip()
+    cipher_name = to_text(b_tmp_header[2].strip())
+    vault_id = default_vault_id
+
+    # Only attempt to find vault_id if the vault file is version 1.2 or newer
+    # if self.b_version == b'1.2':
+    if len(b_tmp_header) >= 4:
+        vault_id = to_text(b_tmp_header[3].strip())
+
+    b_ciphertext = b''.join(b_tmp_data[1:])
+
+    return b_ciphertext, b_version, cipher_name, vault_id
+
+
+def _unhexlify(b_data):
+    try:
+        return unhexlify(b_data)
+    except (BinasciiError, TypeError) as exc:
+        raise AnsibleVaultFormatError(f'Vault format un-hexlify error: {exc}')
```

Comment:
Looks like this is causing tests to fail:

```suggestion
        raise AnsibleVaultFormatError(f'Vault format unhexlify error: {exc}')
```

Reply:
Weird that this would make tests fail on some platforms, but not others.  
Thanks fir the hint, though!

Reply:
I think it failed just for all the platforms running the `ansible-vault` tests because it greps for this particular error.